torch/export/_trace.py

```diff
@@ -956,6 +1013,9 @@ def forward(self, *args, **kwargs):
                     wrapped_mod, new_preserved_call_signatures, module_call_specs
                 ):
                     gm, sig = aot_export(wrapped_mod, args, kwargs=kwargs, **flags)
+                    for buf, name in gm._assigned_buffers.items():
```

Comment:
nit:

Today, the contract for `aot_export_module` is that it returns a tuple of `[GraphModule, GraphSignature]` (where the GraphSignature metadata contains any info about the GraphModule that someone who wants to call it needs to know, like any calling convention changes or input mutations).

I think it would be nice to maintain that invariant, instead of maintaining some side-car extra metadata directly through the `GraphModule`. That would require adding more fields to the `GraphSignature` class here: https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/schemas.py#L550 - but I could also buy that this is kind of heavyweight.

Reply:
Yeah this was ugly, fixed it now by modifying the metadata at the point when buffer assignments are detected, and letting the signature computation do its thing based on the metadata. Didn't need any new fields.